docs/development/ceps/proposed/cep-002.rst

```diff
@@ -0,0 +1,95 @@
+.. _cep-002:
+
+
+***********************
+CEP 2 - Event Structure
+***********************
+
+* Status: draft
+* Discussion: NA
+* Date accepted: NA
+* Last revised: 2023-04-04
+* Author: Maximilian Linhoff
+* Created: 2023-04-04
+
+Abstract
+========
+
+Currently, the hierarchy of the ctapipe ``ArrayEvent`` container has the data
+levels first and then each data level has a ``Map`` ``tel`` for telescope specific
+information in subcontainers.
+This CEP proposes to change this structure to have a container ``TelescopeEvent``
+as parent for all telescope-wise information that then contains the data levels.
+
+Proposed new structure
+======================
+
+This CEP proposes to change the current layout of ``ArrayEvent`` from having multiple
+data levels each with a ``Map`` containing telescope-wise data to a structure
+where each ``ArrayEvent`` is composed of one to many ``TelescopeEvents`` containing
+all telescope-wise information for all data levels.
+
+The ``ArrayEvent`` should also be renamed to ``SubarrayEvent``, to match with other naming
+patterns in ctapipe, such as the ``SubarrayDescription``, making it clear that the array is split
+into multiple subarrays, each observing their own observation block.
+
+The main structure after the change will look like this:
+
+.. code-block::
+
+   SubarrayEvent
+   - index
+   - trigger
+   - simulation
+   - dl0
```

Comment:
What are these "dlX" things? For example the current `DL0Container` looks to mostly just have a telescope-wise map of `DL0CameraContainer`s, but the proposal is to put every telescope-wise thing into a telescope container, right? 

So does the fact that you wrote dlX in both the SubarrayEvent and the TelescopeEvent mean that the current containers are going to be split somehow?

Reply:
The `DLXContainer` s would lose the `.tel` attribute and only store array wide information. The current *CameraContainers will be renamed to *TelescopeContainer and be subcontainers of the `TelescopeEventContainer`.

If there is no array wide information (currently) at a specific data level, this container would no longer exist. But that is also connected to the issue that we currently have some information not sorted into data levels that should be (e.g. trigger should probably be in DL0 and then there is DL0 / subarray / trigger